scripts/comparator/utils/compare_utilities.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"go.uber.org/zap"
+)
+
+const queryFmt string = "/api/v1/query_range?start=%d&end=%d&query=%s&step=%s"
+
+// InputQueries is a slice of InputQuery.
+type InputQueries []InputQuery
+
+// InputQuery is the JSON representation of a query to be compared.
+type InputQuery struct {
+	// QueryGroup is the general category for these queries.
+	QueryGroup string `json:"queryGroup"`
+	// Queries is the list of raw queries.
+	Queries []string `json:"queries"`
+	// Steps is the list of step sizes for these queries.
+	Steps []string `json:"steps"`
+}
+
+// PromQLQueryGroup is a list of constructed PromQL query groups.
+type PromQLQueryGroup struct {
+	// QueryGroup is the general category for these queries.
+	QueryGroup string
+	// Queries is a list of PromQL compatible queries.
+	Queries []string
+}
+
+func (q InputQueries) constructPromQL(
+	start int64,
+	end int64,
+) []PromQLQueryGroup {
+	queries := make([]PromQLQueryGroup, 0, len(q))
+	for _, inQuery := range q {
+		queries = append(queries, inQuery.constructPromQL(start, end))
+	}
+
+	return queries
+}
+
+func (q InputQuery) constructPromQL(start int64, end int64) PromQLQueryGroup {
+	queries := make([]string, 0, len(q.Queries)*len(q.Steps))
+	for _, inQuery := range q.Queries {
+		for _, inStep := range q.Steps {
+			q := strings.ReplaceAll(inQuery, " ", "+")
+			query := fmt.Sprintf(queryFmt, start, end, q, inStep)
```

Comment:
nit: Think it's better to use `url.Values` instead of replacing " " with "+" yourself yeah?  `url.Values` is meant to be the canonical way to encode a query string (and deals with spaces, etc):
https://golang.org/pkg/net/url/#Values.Encode

Should be able to do something just like:
```go
values := make(url.Values)
values.Add("query", inQuery)
values.Add("step", inStep)
values.Add("start", strconv.Itoa(int(start)))
values.Add("end", strconv.Itoa(int(end)))
url := "/api/v1/query_range?" + values.Encode()
```

Reply:
Nice 👍